### DISCUSSION TITLE
LLVM Pipe Rendering On a Single Core?

### DISCUSSION THREAD

**Discussion by tfinnegan937** [pal1000/mesa-dist-win#206] [Q&A] [answered]
Good Morning.

I need to test the performance of an OpenGL application on a single core using software rendering for a high-power embedded target. We would like to quickly set up an LLVM pipe on Windows to test the feasibility of this before we port to our target. I know that OpenGL contexts are usually single-threaded on the CPU side -- Is the LLVM pipe single-threaded as well, or does it thread the back-end? Is it possible to force it to use a single thread?  If we were to go with software rendering on our target, that target would be bound to a single core, so we need to replicate that.

I wholly expect a significant performance hit doing this. I can't go into specifics, but this is a target of last resort.

**Comment by StefanG3D** [comment, marked as answer]
Check out:
https://docs.mesa3d.org/envvars.html#envvar-LP_NUM_THREADS

**Reply by tfinnegan937** [reply]
Perfect, thank you!
